From 831fba504e882aa99e52d923526fe7aeb10f49ef Mon Sep 17 00:00:00 2001
From: Rajani Ranjan <rajani.ranjan@intel.com>
Date: Tue, 22 Apr 2025 06:50:43 +0000
Subject: [PATCH] Porting patch to skip print job if print feature is not
 defined

System doesn't enable the feature "android.software.print",
so we skip the print job.

Test: run monkey test, no crash happen

Tracked-On: OAM-123126
Signed-off-by: Xu Bing <bing.xu@intel.com>
---
 src/com/android/bips/ImagePrintActivity.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/com/android/bips/ImagePrintActivity.java b/src/com/android/bips/ImagePrintActivity.java
index 75e4d2f..5aa166c 100644
--- a/src/com/android/bips/ImagePrintActivity.java
+++ b/src/com/android/bips/ImagePrintActivity.java
@@ -19,6 +19,7 @@ package com.android.bips;
 import android.app.Activity;
 import android.content.Context;
 import android.content.Intent;
+import android.content.pm.PackageManager;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
 import android.graphics.Canvas;
@@ -145,7 +146,8 @@ public class ImagePrintActivity extends Activity {
             boolean isPortrait = values[0];
             if (DEBUG) Log.d(TAG, "startPrint(portrait=" + isPortrait + ")");
             PrintManager printManager = (PrintManager) getSystemService(Context.PRINT_SERVICE);
-            if (printManager == null) {
+	     boolean isSupportPrinting = getPackageManager().hasSystemFeature("android.software.print");
+            if (printManager == null || !isSupportPrinting) {
                 finish();
                 return;
             }
-- 
2.34.1

